components/RuleDisputeForm/RuleDisputeForm.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import { useState } from 'react';
+
+// function App() {
+//   // Similar to useState but first arg is key to the value in local storage.
+//   const [name, setName] = useLocalStorage('name', 'Bob');
+//
+//   return (
+//     <div>
+//       <input
+//         type="text"
+//         placeholder="Enter your name"
+//         value={name}
+//         onChange={e => setName(e.target.value)}
+//       />
+//     </div>
+//   );
+// }
+
+
+
+const RuleDisputeForm = () => {
+  const [expansion, setExpansion] = useLocalStorage('expansion', 'setExpansion')
+  // const [number, setNumber] = useLocalStorage('number', 'setNumber')
+    return(
+      <section>
+      <form>
+        <label>Expansion Name: </label>
+        <input
+        type='text'
+        name='expansion'
+        value={expansion}
+        onChange={event =>setExpansion(event.target.value)}
+        />
+        <label>Number of players: </label>
+        <input
+        type='text'
+        name='number'
+        value={number}
+        onChange={event =>setNumber(event.target.value)}
+        />
+        <label>Gameplay time: </label>
+        <input
+        type='time'
+        name='time'
+        />
+        <label>Official Rule: </label>
+        <input
+        type='text'
+        name='officialRule'
+        />
+        <label>Interpretation of Rule: </label>
+        <input
+        type='text'
+        name='InterpretationRule'
+        />
+
+        <button>SUBMIT</button>
+      </form>
+      </section>
+  )}
+
+  // function useLocalStorage(key, initialValue) {
+  // // State to store our value
+  // // Pass initial state function to useState so logic is only executed once
+  // const [storedValue, setStoredValue] = useState(() => {
+  //   try {
+  //     // Get from local storage by key
+  //     const item = window.localStorage.getItem(key);
+  //     // Parse stored json or if none return initialValue
+  //     return item ? JSON.parse(item) : initialValue;
+  //   } catch (error) {
+  //     // If error also return initialValue
+  //     console.log(error);
+  //     return initialValue;
+  //   }
+  // });
+
+function useLocalStorage(key, initialValue) {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key)
+      return item ? JSON.parse(item): initialValue
+    } catch (error) {
+      console.log(error)
+      return initialValue;
+    }
+  })
+}
```

Comment:
This is rather interesting I would like to know more about this even though we had already looked up this